.changeset/wicked-pants-cough.md

```diff
@@ -0,0 +1,10 @@
+---
+"@sumup-oss/circuit-ui": minor
+---
+
+- Updated the Modal component by using the native `<dialog/>` element and Google Chrome's [dialog-polyfill](https://github.com/GoogleChrome/dialog-polyfill).
+- Modal components can now be invoked directly in your JSX and continue to be invoked via the `useModal` hook with the same interface.
+- Updated the NotificationModal to be based on the new Modal component.
+- The `CloseButtonLabel` is now optional in both components and defaults to "Close", translations are available.
+- Deprecated the `hideCloseButton` prop in both components, previously passed to react-modal's dialog component.
+- A minimum typescript version of 4.1 is now required to use the Modal component.
```

Comment:
Changesets shouldn't contain bullet points since each changeset is turned into a list item in the changelog. Multiple change descriptions should be turned into separate changesets.

The `dialog-polyfill` dependency is an implementation detail.

```suggestion
Refactored the Modal component to use the native `dialog` element. The Modal component can now be rendered directly in your JSX (the older `useModal` hook continues to be supported).
```

Separate changesets:

```
Refactored the NotificationModal component to use the new Modal component under the hood.
```

```
Added default translations for the Modal and NotificationModal components. The `closeButtonLabel` prop is now optional.
```

```
Deprecated the `hideCloseButton` prop in the Modal and NotificationModal components. It had no effect.
```

```
Set an explicit minimum version for TypeScript of 4.1 or higher. While this is technically a breaking change, v4.1 was released over 4 years ago, so we don't expect this to break anyone's code. Please let us know if this causes you issues.
```

TypeScript v4.1+ is required no matter if the Modal component is imported since TypeScript parses all modules in the module graph regardless.